sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetryMetrics.java

```diff
@@ -0,0 +1,907 @@
+                .register(compositeRegistry);
+            channelTaskQueueSizeMeter.record(channelTaskQueueSize);
+        }*/
+
+        private void recordRntbdEndpointStatistics(RntbdEndpointStatistics endpointStatistics, Tags requestTags) {
+            if (endpointStatistics == null) {
+                return;
+            }
+
+            DistributionSummary acquiredChannelsMeter = DistributionSummary
+                .builder(nameOf("req.rntbd.stats.endpoint.acquiredChannels"))
+                .baseUnit("#")
+                .description("Endpoint statistics(acquired channels)")
+                .maximumExpectedValue(100_000d)
+                .publishPercentiles()
+                .publishPercentileHistogram(false)
+                .tags(requestTags)
+                .register(compositeRegistry);
+            acquiredChannelsMeter.record(endpointStatistics.getAcquiredChannels());
+
+            DistributionSummary availableChannelsMeter = DistributionSummary
+                .builder(nameOf("req.rntbd.stats.endpoint.availableChannels"))
+                .baseUnit("#")
+                .description("Endpoint statistics(available channels)")
+                .maximumExpectedValue(100_000d)
+                .publishPercentiles()
+                .publishPercentileHistogram(false)
+                .tags(requestTags)
+                .register(compositeRegistry);
+            availableChannelsMeter.record(endpointStatistics.getAvailableChannels());
+
+            DistributionSummary inflightRequestsMeter = DistributionSummary
+                .builder(nameOf("req.rntbd.stats.endpoint.inflightRequests"))
+                .baseUnit("#")
+                .description("Endpoint statistics(inflight requests)")
+                .tags(requestTags)
+                .maximumExpectedValue(1_000_000d)
+                .publishPercentiles(0.95, 0.99)
+                .publishPercentileHistogram(true)
+                .register(compositeRegistry);
+            inflightRequestsMeter.record(endpointStatistics.getInflightRequests());
+
+            // TODO @fabianm - delete completely if we agree in PR that this metric isn't needed
+            /*
+            DistributionSummary executorTaskQueueSizeMeter = DistributionSummary
+                .builder(nameOf("req.rntbd.stats.endpoint.executorTaskQueueSize"))
+                .baseUnit("#")
+                .description("Endpoint statistics(executor task queue size)")
+                .publishPercentiles()
+                .publishPercentileHistogram(false)
+                .tags(requestTags)
+                .register(compositeRegistry);
+            executorTaskQueueSizeMeter.record(endpointStatistics.getExecutorTaskQueueSize());*/
+        }
+
+        // TODO @fabianm - delete completely if we agree in PR that this metric isn't needed
+        /*
+        private void recordRntbdChannelAcquisitionTimeline(
+            RntbdChannelAcquisitionTimeline acquisitionTimeline,
```

Comment:
it might be useful to keep the transit time and channelAcquisitionTime

Reply:
those are kept - this is just for the sub steps during channel acquisition that you recently added - but keeping them in diagnostics should be sufficient IMO